internal/error.go

```diff
@@ -9,10 +10,10 @@ import (
 )
 
 func IsRetryableError(err error, retryTimeout bool) bool {
-	if err == nil {
+	switch err {
+	case nil, context.Canceled, context.DeadlineExceeded:
```

Comment:
Is this needed? I would expect `context.Canceled|DeadlineExceeded` to return false in the previous code

Reply:
It is not needed, but `err == context.Canceled` is fast(er) and it is also explicit.